src/salus.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2023 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// Salus vendor exception.
+
+use crate::error::*;
+use crate::vendor::*;
+use crate::SbiFunction;
+
+const EXT_SALUS_TEST: u64 = 0x09FFFFFF;
+
+pub type SalusSbiMessage = VendorSbiMessage<SalusExtension>;
+
+pub enum SalusExtension {
```

Comment:
Why is SalusExtension an enum? This extension is for a7 == 0x09FFFFFF (Test), any other extensions added will need a different a7 and be a different extension, just like another vendors specific extension. Or am I missing something?

Reply:
Ah, that makes sense. Maybe "SalusExtenions" plural then?

For Rivos we'll likely have different but overlapping sets of extensions for m-mode FW, s-mode FW, and Salus. How would those be shared? I'm considering the trade off between adding one file per extension and one file per SBI implementation. Thoughts?

Reply:
So if they're different but overlapping I guess this is kind of what I was going for, at least for now.
Anyone using `sbi-rs` doesn't have to use the salus extensions. Has to import the module manually.
Probably the code will have to be restructured in the future, but for now I was thinking of keeping all salus functions for all salus extensions in that file, and then possibly move them in a module with its own directory (salus/mod.rs) when it grows out of a sane single file source.
There's also the possibility for the user of `sbi-rs` would implement the set of extensions outside of `sbi-rs`, and then how it implements that will be project specific.

Reply:
sgtm. @tdewey-rivos does this work for you? If so, we'll merge this one, nothing set in stone, still at 0.1 so APIs can evolve as needed.

Reply:
am about to push some changes to this PR allow API calls.